crates/bgp-pkt/src/iana.rs

```diff
@@ -1354,6 +1354,396 @@ impl TryFrom<u8> for AigpAttributeType {
+}
+
+impl TryFrom<u16> for BgpLsAttributeTlv {
+    type Error = UnknownBgpLsAttributeTlvType;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        match Self::from_repr(value) {
+            Some(val) => Ok(val),
+            None => Err(UnknownBgpLsAttributeTlvType(value)),
+        }
+    }
+}
+
+#[repr(u16)]
+#[derive(Display, FromRepr, Copy, Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
+pub enum BgpLsNodeAttribute {
+    MultiTopologyIdentifier = 263,
+    NodeFlagBits = 1024,
+    OpaqueNodeAttribute = 1025,
+    NodeNameTlv = 1026,
+    IsIsArea = 1027,
+    LocalNodeIpv4RouterId = 1028,
+    LocalNodeIpv6RouterId = 1029,
+}
+#[repr(u16)]
+#[derive(Display, FromRepr, Copy, Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
+pub enum BgpLsLinkAttribute {
+    LocalNodeIpv4RouterId = 1028,
+    LocalNodeIpv6RouterId = 1029,
+    RemoteNodeIpv4RouterId = 1030,
+    RemoteNodeIpv6RouterId = 1031,
+    RemoteNodeAdministrativeGroupColor = 1088,
+    MaximumLinkBandwidth = 1089,
+    MaximumReservableLinkBandwidth = 1090,
+    UnreservedBandwidth = 1091,
+    TeDefaultMetric = 1092,
+    LinkProtectionType = 1093,
+    MplsProtocolMask = 1094,
+    IgpMetric = 1095,
+    SharedRiskLinkGroup = 1096,
+    OpaqueLinkAttribute = 1097,
+    LinkName = 1098,
+}
+
+#[repr(u16)]
+#[derive(Display, FromRepr, Copy, Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
+pub enum BgpLsPrefixAttribute {
+    IgpFlags = 1152,
+    IgpRouteTag = 1153,
+    IgpExtendedRouteTag = 1154,
+    PrefixMetric = 1155,
+    OspfForwardingAddress = 1156,
+    OpaquePrefixAttribute = 1157,
+}
+
+#[repr(u8)]
+#[derive(Display, FromRepr, Copy, Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
+/// ```text
+///        0 1 2 3 4 5 6 7
+///       +-+-+-+-+-+-+-+-+
+///       |V|L|B|P| Rsvd  |
+///       +-+-+-+-+-+-+-+-+
+/// ```
+/// - V-Flag: Value Flag.  If set, then the SID carries a label value.  By
+///   default, the flag is SET.
+///
+/// - L-Flag: Local Flag.  If set, then the value/index carried by the SID has
+///   local significance.  By default, the flag is SET.
+///
+/// - B-Flag: Backup Flag.  If set, the SID refers to a path that is eligible
+///   for protection using fast reroute (FRR).  The computation of the backup
+///   forwarding path and its association with the BGP Peering SID forwarding
+///   entry is implementation specific.  Section 3.6 of [RFC9087] discusses some
+///   of the possible ways of identifying backup paths for BGP Peering SIDs.
+///
+/// - P-Flag: Persistent Flag: If set, the SID is persistently allocated, i.e.,
+///   the SID value remains consistent across router restart and
+///   session/interface flap.
+///
+/// - Rsvd bits: Reserved for future use and MUST be zero when originated and
+///   ignored when received.
+pub enum BgpLsSidAttributeFlags {
+    ValueFlag = 0b_1000_0000,
+    LocalFlag = 0b_0100_0000,
+    BackupFlag = 0b_0010_0000,
+    PersistentFlag = 0b_0001_0000,
+}
+
+#[repr(u8)]
```

Comment:
for consistent styling add `#[repr(u8)]`, `#[derive` directly before the type definition and after the comments.